metricbeat/scripts/mage/docsCollector.go

```diff
@@ -0,0 +1,488 @@
+
+		isModule, err := testIfDocsInDir(module)
+		if err != nil {
+			return moduleMap, err
+		}
+		if !isModule {
+			continue
+		}
+		moduleName := filepath.Base(module)
+
+		err = createDocsPath(moduleName)
+		if err != nil {
+			return moduleMap, err
+		}
+
+		fieldsm, err := loadModuleFields(filepath.Join(module, "_meta/fields.yml"))
+		if err != nil {
+			return moduleMap, err
+		}
+
+		cfgPath, err := getConfigfile(module)
+		if err != nil {
+			return moduleMap, err
+		}
+
+		metricsets, err := gatherMetricsets(module, moduleName)
+		if err != nil {
+			return moduleMap, err
+		}
+
+		//dump the contents of the module asciidoc
+		moduleDoc, err := ioutil.ReadFile(filepath.Join(module, "_meta/docs.asciidoc"))
+		if err != nil {
+			return moduleMap, err
+		}
+
+		fieldsm.Path = module
+		fieldsm.CfgFile = cfgPath
+		fieldsm.Metricsets = metricsets
+		fieldsm.Asciidoc = string(moduleDoc)
+		fieldsm.Dashboards = hasDashboards(module)
+
+		moduleMap[moduleName] = fieldsm
+
+	} // end of modules loop
+
+	return moduleMap, nil
+}
+
+// write the module-level docs
+func writeModuleDocs(modules map[string]moduleData) error {
+	for moduleName, mod := range modules {
+
+		moduleFile := generatedNote
+		moduleFile = moduleFile + fmt.Sprintf("[[metricbeat-module-%s]]\n", moduleName)
+		moduleFile = moduleFile + fmt.Sprintf("== %s module\n\n", mod.Title)
+
+		if mod.Release != "ga" {
+			moduleFile = moduleFile + fmt.Sprintf("%s[]\n\n", mod.Release)
+		}
+
+		//Add the asciidoc lines and config header
+		moduleFile = moduleFile + mod.Asciidoc + fmt.Sprintf(moduleExampleConfig, mod.Title)
+		//Add the config
+		moduleFile = moduleFile + mod.CfgFile + "----\n\n"
+
+		//we're doing this in a somewhat klunky way to insure the order of the original python script is preserved
+		additonalHelpers := false
+		if mod.checkConfig("ssl") {
+			moduleFile = moduleFile + "This module supports TLS connections when using `ssl` config field, as described in <<configuration-ssl>>.\n"
+			additonalHelpers = true
+		}
+		if mod.checkConfig("http") {
+			moduleFile = moduleFile + "It also supports the options described in <<module-http-config-options>>.\n"
+			additonalHelpers = true
+		}
+
+		if additonalHelpers {
+			moduleFile = moduleFile + "\n"
+		}
+
+		//Add the metricset links
+		moduleFile = moduleFile + "[float]\n=== Metricsets\n\nThe following metricsets are available:\n\n"
+		//iterate over the metricsets, adding links and includes.
+		//Again, this particular way is done to preserve the output of the original python script
+		for _, ms := range mod.Metricsets {
+			moduleFile = moduleFile + fmt.Sprintf("* %s\n\n", ms.Link)
+		}
+		for _, ms := range mod.Metricsets {
+			moduleFile = moduleFile + fmt.Sprintf("include::%s/%s.asciidoc[]\n\n", moduleName, ms.Title)
+		}
+
+		//write to doc file
+		filename := mage.OSSBeatDir(filepath.Join("docs", "modules", fmt.Sprintf("%s.asciidoc", moduleName)))
+		err := ioutil.WriteFile(filename, []byte(moduleFile), 744)
```

Comment:
This creates the files with weird permisions:
```
--wxr-x--- 1 jaime jaime 904 jul  1 10:11 docs/modules/aerospike.asciidoc
```
I think this should be
```suggestion
		err := ioutil.WriteFile(filename, []byte(moduleFile), 0644)
```

And the same in other places where `744` is used as mode.